core/config.go

```diff
@@ -79,3 +77,13 @@ func loadFromEnv(configMap *koanf.Koanf) error {
 func loadFromFlagSet(configMap *koanf.Koanf, flags *pflag.FlagSet) error {
 	return configMap.Load(posflag.Provider(flags, defaultDelimiter, configMap), nil)
 }
+
+// splitWithEscaping see https://codereview.stackexchange.com/questions/259270/golang-splitting-a-string-by-a-separator-not-prefixed-by-an-escape-string/259382
```

Comment:
We (can do)[https://codereview.stackexchange.com/a/280193/265515] better than that. 🤓

Reply:
Doesn't work;

```golang
        	Error Trace:	nuts-node/core/config_test.go:117
        	Error:      	Not equal: 
        	            	expected: []string{"", "a", "b", "c", "d", "value-,-with-escaped-comma", ""}
        	            	actual  : []string{"", "a", "b", "c", "d", "value-\\,-with-escaped-comma", ""}
```

Reply:
If you want to trim the escaped, then don't add it back. That is:

    a[i] += separator + a[i+1]

... becomes:

    a[i] += a[i+1]

Reply:
acutally:

```golang
func splitWithEscaping(s, separator, escape string) []string {
	a := strings.Split(s, separator)

	for i := len(a) - 2; i >= 0; i-- {
		if strings.HasSuffix(a[i], escape) {
			a[i] = a[i][:len(a[i])-1] + separator + a[i+1]
			a = append(a[:i+1], a[i+2:]...)
		}
	}
	return a
}
```

Reply:
But hey, it's only executed once, and algorithms with reslicing in a loop isn't necessarily more readable than a magic string replace, so potato-potato.